core-rust/mesh-api-server/src/mesh_api/types.rs

```diff
@@ -0,0 +1,38 @@
+use crate::engine_prelude::*;
+use crate::prelude::*;
+
+#[derive(Debug, Clone, EnumIter, Display, FromRepr)]
+#[repr(i64)]
+pub(crate) enum MeshApiOperationTypes {
+    Withdraw,
+    Deposit,
+    // LockFee,
```

Comment:
In the design doc, I suggested the following. These allow us to ensure that the operations in a constructed transaction are a subset of the operations in a returned transaction.

In particular `UnusedLockFee` is refunded; and a non-contingent `LockFee` and `UnusedLockFee` would always have an operation success of success for a committed transaction.
```
Withdraw - against any Global Entity
Deposit - against any Global Entity
Mint (Because mint events by themselves don't have a clear destination; I suggest a mint event is converted to an equal “mint” to the resource itself, and then an equal “withdraw” from the resource)
Burn (Similarly, a burn event is converted to an equal “burn” and then a “deposit” against the resource itself)
LockFee (We split fee burn into the “lock fee” operation from the intent / event, and the unused lock fee refund)
UnusedLockFee
```